pkg/controller/argocd/policyrule.go

```diff
@@ -126,6 +139,17 @@ func policyRuleForDexServer() []v1.PolicyRule {
 
 func policyRuleForServer() []v1.PolicyRule {
 	return []v1.PolicyRule{
+		{
+			APIGroups: []string{
```

Comment:
And pls explain why this is added.

Reply:
For the same reason as described for the `application-controller`

Reply:
Do we need to move this here?
```
{
			APIGroups: []string{
				"",
			},
			Resources: []string{
				"pods",
				"pods/log",
			},
			Verbs: []string{
				"get",
			},
		},
        ```

Reply:
Also, we should merge the rules of the same api group.

Reply:
We don't need to add this, since it has the permissions to read all the resources in all API groups within the namespace. So the existing rule takes care of this!